main/src/library/ChannelImpl.flix

```diff
@@ -0,0 +1,344 @@
+namespace ChannelImpl {
+
+    /**
+    * The tuple corresponds to (id, channelLock, bufferType, elementQueue, waitingGetters, waitingSetters)
+    *   - id
+    *   - channelLock is the channelLock of this channel.
+    *   - bufferSize is the size of a channel or unbuffered. If you try to put an element in a channel that's full, you wait until there's space.
+    *   - elementQueue is a circular buffer of elements in the list.
+    *   - waitingGetters is a set of conditions that is waiting for get. This set is cleared after each new element.
+    *   - waitingSetters is a condition that can notify threads of available space in the elementQueue.
+    */
+    pub enum ChannelImpl[a] {
+        case Mpmc(
+                Int64,
+                ReentrantLock,
+                Int,
+                CircularBuffer[a],
+                MutList[(ReentrantLock, Condition)],
+                Condition
+            )
+    }
+
+    pub def newWithId(bufferSize: Int): ChannelImpl[a] & Impure =
+        //import dev.flix.runtime.GlobalCounter:newId();
```

Comment:
@magnus-madsen I couldn't get this to work, does it need to be added in the compiler somewhere?

Reply:
Flix tries to load the class, but I guess it does not exist yet!

Reply:
We need to change the interop to support this notion of separate compilation. 

A temporary hack could be to use a random int64

Reply:
Do you think its solvable now or is it dependent on other things?

Reply:
You can solve it, but it will require some work. I can write up a ticket?

Reply:
yea sure, that sounds good